ZKLND-39 add test cases and check channel status

DariusParvin opened the pull request:
#### Pull Request Checklist

- [ ] If this is your first time contributing, we recommend you read the [Code
  Contribution Guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md)
- [ ] All changes are Go version 1.12 compliant
- [ ] The code being submitted is commented according to [Code Documentation and Commenting](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#CodeDocumentation)
- [ ] For new code: Code is accompanied by tests which exercise both
  the positive and negative (error paths) conditions (if applicable)
- [ ] For bug fixes: Code is accompanied by new tests which trigger
  the bug being fixed to prevent regressions
- [ ] Any new logging statements use an appropriate subsystem and
  logging level
- [ ] Code has been formatted with `go fmt`
- [ ] Protobuf files (`lnrpc/**/*.proto`) have been formatted with
  `make rpc-format` and compiled with `make rpc`
- [ ] For code and documentation: lines are wrapped at 80 characters
  (the tab character should be counted as 8 characters, not 4, as some IDEs do
  per default)
- [ ] Running `make check` does not fail any tests
- [ ] Running `go vet` does not report any issues
- [ ] Running `make lint` does not report any **new** issues that did not
  already exist
- [ ] All commits build properly and pass tests. Only in exceptional
  cases it can be justifiable to violate this condition. In that case, the
  reason should be stated in the commit message.
- [ ] Commits have a logical structure according to [Ideal Git Commit Structure](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#IdealGitCommitStructure)

DariusParvin committed:
fix db path

DariusParvin committed:
fix db path

DariusParvin committed:
mark channels as ConfirmedClosed

DariusParvin committed:
mark channels as ConfirmedClosed

DariusParvin committed:
ZKLND-39 add test cases to zkchainwatcher

- Add test cases to zkchainwatcher
- moved setupLibzkChannels to test_utils. (first step to moving zkchannel tests to it's own package)
- need to add tests for remoteValidCustClose and remoteInvalidCustClose

DariusParvin committed:
ZKLND-39 add custClose test scenarios

DariusParvin committed:
move global test constants to zkchannels test_utils functions

DariusParvin committed:
ZKLND-39 Add checks to channel status when remote closing tx is detected

gijsvl commented on zkchannelmanager.go:
```diff
@@ -1234,6 +1221,19 @@ func (z *zkChannelManager) advanceMerchantStateAfterConfirmations(notifier chain
 		zkchLog.Error(err)
 	}
 
+	if confirmOpen == true {
```
Redundant `== true`

gijsvl replied:
There is one on line 1184 as well

gijsvl commented on go.mod:
```diff
@@ -57,6 +57,8 @@ require (
 	github.com/miekg/dns v0.0.0-20171125082028-79bfde677fa8
 	github.com/modern-go/reflect2 v1.0.1 // indirect
 	github.com/niemeyer/pretty v0.0.0-20200227124842-a10e7caefd8e // indirect
+	github.com/onsi/ginkgo v1.7.0 // indirect
```
I think they need to be removed, how did they appear again? I changed some small things in `.gitignore` and `go.mod` to solve some small problems, but these lines were removed... Try and run `go mod tidy` on your side.

gijsvl commented on zkchannels/test_utils.go:
```diff
@@ -0,0 +1,442 @@
+	if err != nil {
+		log.Fatalf("%v", err)
+	}
+	_ = signedEscrowTx
```
why?

gijsvl commented on zkchannels/test_utils.go:
```diff
@@ -0,0 +1,442 @@
+	if err != nil {
+		log.Fatalf("%v", err)
+	}
+	_ = merchTxid_LE
```
also why?

gijsvl commented on zkchannels/test_utils.go:
```diff
@@ -0,0 +1,442 @@
+		InitMerchBal:  merchBal,
+	}
+
+	custClosePk := custState.PayoutPk
```
Why the reassignment?

DariusParvin committed:
Update go.mod

remove github.com/onsi/ginkgo v1.7.0 // indirect

DariusParvin committed:
minor edits

DariusParvin committed:
add tickets for todos

DariusParvin committed:
remove comment

DariusParvin committed:
ZKLND-53  store sessionID on customer's side during pay

This way, the customer can identify which channel the merchant is communicating on based on the sessionID. Previously the customer may have run into problems if a second payment was initiated while another was in progress.

DariusParvin committed:
ZKLND-33 add generate fresh pubkey using lnwallet

add lnwallet field to zkchannelmanager, zkchainwatcher, and zkbreacharbiter.

use lnwallet.NewPubKey() in zkchainwatcher and zkbreacharbiter when creating closing txs.

DariusParvin committed:
clean up
